storage: Send node liveness heartbeats more frequently

a-robinson opened the pull request:
Also log when they take more than a second.

Hopefully the first commit will reduce the frequency of blips in the per-node view of node liveness that we've been seeing in our metrics.

I'm not sure the second commit is worth it -- if a lot of heartbeats get backed up it could seemingly generate a fair amount of log spam. Let me know what you think.

@spencerkimball

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/12500)
<!-- Reviewable:end -->

a-robinson committed:
storage: Send node liveness heartbeats more frequently

a-robinson committed:
storage: Log when node liveness heartbeat takes more than a second

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r1.
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[pkg/storage/node_liveness.go, line 218 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRrUxmTa6Ug60nT_Hg:-KZRrUxmTa6Ug60nT_Hh:bqeh5wq) ([raw file](https://github.com/cockroachdb/cockroach/blob/3757a26fdbb05a312d52f13dab072c2a5abea2e2/pkg/storage/node_liveness.go#L218)):*
> ```Go
> 	defer func() {
> 		if dur := timeutil.Now().Sub(start); dur > time.Second {
> 			log.Warningf(ctx, "slow heartbeat took %v", dur)
> ```

I like printing durations in seconds to make them sortable when grep'ing. `%0.1fs` and `dur.Seconds()`.

---

*[pkg/storage/node_liveness.go, line 220 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRrMH2U9smpVFeFYGe:-KZRrMH30rxvJ2G3dAKU:b-g7r3rn) ([raw file](https://github.com/cockroachdb/cockroach/blob/3757a26fdbb05a312d52f13dab072c2a5abea2e2/pkg/storage/node_liveness.go#L220)):*
> ```Go
> 			log.Warningf(ctx, "slow heartbeat took %v", dur)
> 		}
> 	}()
> ```

```
defer func(start time.Time) {
}(timeutil.Now())
```

---

*[pkg/storage/store.go, line 89 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRr1aLKH9x_fMgfTLx:-KZRr1aLKH9x_fMgfTLy:b-ytdxjr) ([raw file](https://github.com/cockroachdb/cockroach/blob/704d5d37eed4c5442f6123c7a4c2521c578f07d4/pkg/storage/store.go#L89)):*
> ```Go
> 	// livenessRenewalDivisor specifies what quotient the node liveness renewal
> 	// duration should be of the node liveness duration.
> 	livenessRenewalDivisor = 2
> ```

Can we make this `livenessRenewalFraction = 0.5`? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-:-KZRrtFkSRoSRhamMhBH:bnfp4nl)*
<!-- Sent from Reviewable.io -->

a-robinson commented:
Review status: 1 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[pkg/storage/node_liveness.go, line 218 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRrUxmTa6Ug60nT_Hg:-KZRu1hM6z_XNCYF8xWl:bfp2nd) ([raw file](https://github.com/cockroachdb/cockroach/blob/3757a26fdbb05a312d52f13dab072c2a5abea2e2/pkg/storage/node_liveness.go#L218)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I like printing durations in seconds to make them sortable when grep'ing. `%0.1fs` and `dur.Seconds()`.
</blockquote></details>

That's right, sorry for forgetting. Changed.

---

*[pkg/storage/node_liveness.go, line 220 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRrMH2U9smpVFeFYGe:-KZRu0VcHdUvjMhjO9Id:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/3757a26fdbb05a312d52f13dab072c2a5abea2e2/pkg/storage/node_liveness.go#L220)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

```
defer func(start time.Time) {
}(timeutil.Now())
```
</blockquote></details>

Done.

---

*[pkg/storage/store.go, line 89 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRr1aLKH9x_fMgfTLx:-KZRuGSOKm51LM8Y2pJC:bs03j6n) ([raw file](https://github.com/cockroachdb/cockroach/blob/704d5d37eed4c5442f6123c7a4c2521c578f07d4/pkg/storage/store.go#L89)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Can we make this `livenessRenewalFraction = 0.5`? 
</blockquote></details>

Yeah, I had been thinking while writing this that an integer divisor isn't very flexible but figured I'd stay consistent with the other ones. Changed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/storage/store.go, line 89 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZRr1aLKH9x_fMgfTLx:-KZRvPNPD0pmU1Mh2JMu:b-hbv20o) ([raw file](https://github.com/cockroachdb/cockroach/blob/704d5d37eed4c5442f6123c7a4c2521c578f07d4/pkg/storage/store.go#L89)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Yeah, I had been thinking while writing this that an integer divisor isn't very flexible but figured I'd stay consistent with the other ones. Changed.
</blockquote></details>

I'd be happy to see `rangeLeaseRenewalDivisor` changed to `rangeLeaseRenewalFraction = 0.8`, though you need to be careful about that change. Perhaps for a follow-on PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 2 files at r1, 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/server/server.go, line 215 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500#-KZSQZGdqmL4Qr9FvYHu:-KZSQZGdqmL4Qr9FvYHv:bc8z1iu) ([raw file](https://github.com/cockroachdb/cockroach/blob/7b6e6e11fdd7b3c15f22d51d1f54e234faceaf43/pkg/server/server.go#L215)):*
> ```Go
> 	s.db = client.NewDB(s.txnCoordSender)
> 
> 	// Use the range lease expiration and renewal durations as the node
> ```

this comment is nonsense now, i think.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12500)*
<!-- Sent from Reviewable.io -->

spencerkimball reviewed (commented):
LGTM

a-robinson commented:
So far this has gotten rid of blips in cyan's per-node graphs, although cyan has also been down half the time since this got pushed out so it's not totally conclusive yet.
